src/rpc/cli.cpp

```diff
@@ -114,57 +114,75 @@ void cli::run()
 /****
  * @brief loop through list of commands, attempting to find a match
  * @param token what the user typed
- * @param match
- * @returns the full name of the command or NULL if no matches found
+ * @param match sets to 1 if only 1 match was found
+ * @returns the remaining letters of the name of the command or NULL if 1 match not found
  */
 static char *my_rl_complete(char *token, int *match)
 {
-    int matchlen = 0;
-    int count = 0;
-    const char* method_name;
-
-    auto& cmd = cli_commands();
-    int partlen = strlen (token); /* Part of token */
-    for (auto it : cmd) {
-    	if (!strncmp (it.c_str(), token, partlen)) {
-    		method_name = it.c_str();
-    		matchlen = partlen;
-    		count ++;
-    	}
-    }
-
-    if (count == 1) {
-    	*match = 1;
-    	return strdup (method_name + matchlen);
-    }
-
-    return NULL;
+   bool have_one = false;
+   std::string method_name;
+
+   auto& cmd = cli_commands();
+   int partlen = strlen (token); /* Part of token */
+
+   for (const std::string it : cmd)
+   {
+      if (it.compare(0, partlen, token) == 0)
+      {
+         if (have_one) {
+            // we can only have 1, but we found a second
```

Comment:
Actually I think the function should be able to do partially completion if found more matches. E.G. type `ge`, press tab key, then it would be partially completed to `get` (then list several commands start with `get` with another function). This is the original behavior. However, we can leave this feature to the future if the schedule is too tight.

Reply:
I agree. I'm working out the algorithm...